librad/src/net/gossip/error.rs

```diff
@@ -17,13 +17,14 @@
 
 use std::io;
 
-use futures_codec::CborCodecError;
 use thiserror::Error;
 
+use crate::net::codec::CborCodecError;
+
 #[derive(Debug, Error)]
 pub enum Error {
     #[error("Invalid payload")]
```

Comment:
Ya, it's actually quite annoying: for some reason `futures_codec::Decoder::Error` is bounded on `From<io::Error>` (because it's interleaving IO and decoding, I guess). `serde_cbor` doesn't return any IO errors, so I could tear the error type apart previously. But `minicbor` does (and I don't understand why exactly), so we really get either an IO or decode error here. *sigh* Just calling it a `Error::Cbor(#[from] CborCodecError)` :man_shrugging: